modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -180,18 +183,30 @@ AzureBlobServiceClient createClient(
         final HttpClient httpClient = new NettyAsyncHttpClientBuilder(nettyHttpClient).disableBufferCopy(true).proxy(proxyOptions).build();
 
         final String connectionString = settings.getConnectString();
+        final StorageConnectionString storageConnectionString = StorageConnectionString.create(connectionString, clientLogger);
+        final StorageEndpoint endpoint = storageConnectionString.getBlobEndpoint();
```

Comment:
I believe there an issue with that because the SDK doesn't export these classes in `module-info`, but I'm not sure whether we enforce it.

Reply:
We were using the `StorageConnectionString` class already. I agree that we should move to our own connection string parsing, but we can defer that.